packages/near-membrane-base/src/intrinsics.ts

```diff
@@ -147,13 +154,19 @@ export function getFilteredEndowmentDescriptors(endowments: object): PropertyDes
         // will be ignored if present in the endowments object.
         // TODO: what if the intent is to polyfill one of those
         // intrinsics?
-        if (!ReflectApply(SetProtoHas, ESGlobalKeys, [key])) {
+        if (!isIntrinsicGlobalName(key)) {
             to[key] = ReflectGetOwnPropertyDescriptor(endowments, key)!;
         }
     }
     return to;
 }
 
-export function isIntrinsicGlobalName(key: PropertyKey): boolean {
-    return ReflectApply(SetProtoHas, ESGlobalKeys, [key]);
+export function getResolvedShapeDescriptors(
+    globalObjectShape: typeof globalThis,
+    endowments: object
+): PropertyDescriptorMap {
+    const globalObjectShapeDescriptors = getFilteredGlobalObjectShapeDescriptors(globalObjectShape);
+    const endowmentsDescriptors = getFilteredGlobalObjectShapeDescriptors(endowments);
+    // eslint-disable-next-line prefer-object-spread
+    return ObjectAssign({}, globalObjectShapeDescriptors, endowmentsDescriptors);
```

Comment:
Null proto for the returned object here? I noticed it is nulled in `getFilteredGlobalObjectShapeDescriptors`

Reply:
Oh, good call.

Reply:
```suggestion
    return ObjectAssign({ __proto__: null }, globalObjectShapeDescriptors, endowmentsDescriptors);
```